From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Tue, 10 Feb 2026 19:25:21 +0100
Subject: [PATCH] Fix build with zxing-cpp 3

The pc file no longer adds ${includedir}/ZXing to the include paths.
---
 subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp b/subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp
index 478c678ebb39..931b4fb609f4 100644
--- a/subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp
+++ b/subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp
@@ -62,9 +62,8 @@
 
 #include <gst/video/video.h>
 
-#include "ReadBarcode.h"
-#include "TextUtfEncoding.h"
-#include "ZXVersion.h"
+#include "ZXing/ReadBarcode.h"
+#include "ZXing/Version.h"
 
 using namespace ZXing;
 
@@ -415,7 +414,7 @@ gst_zxing_transform_frame_ip (GstVideoFilter * vfilter, GstVideoFrame * frame)
   GstZXing *zxing = GST_ZXING (vfilter);
   gpointer data;
   gint height, width;
-  #if ZXING_VERSION_MAJOR >= 2 && ZXING_VERSION_MINOR >= 2
+  #if ZXING_VERSION_MAJOR >= 3 || (ZXING_VERSION_MAJOR == 2 && ZXING_VERSION_MINOR >= 2)
     ReaderOptions hints;
   #else
     DecodeHints hints;
